components/brave_wallet/browser/internal/hd_key.cc

```diff
@@ -358,24 +336,21 @@ std::unique_ptr<HDKey> HDKey::GenerateFromV3UTC(const std::string& password,
     return nullptr;
   }
 
-  std::unique_ptr<std::vector<uint8_t>, SecureZeroVectorDeleter<uint8_t>>
-      private_key(new std::vector<uint8_t>(),
-                  SecureZeroVectorDeleter<uint8_t>());
+  std::vector<uint8_t> private_key;
   if (!UTCDecryptPrivateKey(derived_key->key(), ciphertext_bytes, iv_bytes,
-                            private_key.get(), *dklen))
+                            &private_key, *dklen)) {
     return nullptr;
+  }
 
-  return GenerateFromPrivateKey(*std::move(private_key));
+  return GenerateFromPrivateKey(private_key);
```

Comment:
Incoming json and parsed private key vector have same level of memory security requirements, so to make code simple I've chosen to use plain vector here.